src/main/java/org/jooq/lambda/Seq.java

```diff
@@ -9688,15 +9688,8 @@ public T next() {
      * </pre></code>
      */
     static <T> Tuple2<Seq<T>, Seq<T>> splitAt(Stream<? extends T> stream, long position) {
-        return seq(stream)
-            .zipWithIndex()
-            .partition(t -> t.v2 < position)
-            // Explicit type parameters to work around this Eclipse compiler bug:
-            // https://bugs.eclipse.org/bugs/show_bug.cgi?id=455945
-            .map((v1, v2) -> Tuple.<Seq<T>, Seq<T>>tuple(
-                v1.map(t -> t.v1),
-                v2.map(t -> t.v1)
-            ));
+        SeqBuffer<T> buffer = SeqBuffer.of(stream);
+        return tuple(buffer.seq().limit(position), buffer.seq().skip(position));
```

Comment:
There's currently no guarantee in the API, as there are too many flaws, still. But in principle, the results from methods like `splitAt()` (which produce several Seq) should be thread safe. Or at least, we should have an option for them to be thread safe.

Perhaps this is a concept that should be reviewed more globally.

Note that thread safety and parallelism aren't the same thing in this context. Stream's parallelism allows for parallel processing of operations like `map()` or `filter()`. By keeping operations independent of one another, parallelism can drastically speed up the processing of a stream. Thread safety in our context just means that consuming two things (e.g. the `splitAt()` results) on different threads, we don't want to get wrong results.

But again, perhaps we should implement this more thoroughly, with a specific thread safety flag...